BLE Indoor Positioning/src/main/java/com/nexenio/bleindoorpositioning/ble/advertising/AdvertisingPacketFactory.java

```diff
@@ -1,13 +1,111 @@
 package com.nexenio.bleindoorpositioning.ble.advertising;
 
+import java.util.HashMap;
+import java.util.Map;
+
 /**
  * Created by steppschuh on 02.02.18.
  */
 
-public interface AdvertisingPacketFactory {
+public abstract class AdvertisingPacketFactory<AP extends AdvertisingPacket> {
+
+    private Map<Class<AP>, AdvertisingPacketFactory<AP>> subFactoryMap = new HashMap<>();
+    private Class<AP> packetClass;
+
+    public AdvertisingPacketFactory(Class<AP> packetClass) {
+        this.packetClass = packetClass;
+    }
+
+    abstract boolean canCreateAdvertisingPacket(byte[] advertisingData);
+
+    protected boolean canCreateAdvertisingPacketWithDescendants(byte[] advertisingData) {
```

Comment:
Maybe use `SubFactories` instead of `Descendants`, to stay consistent with `subFactoryMap `